client/my-sites/site-settings/settings-newsletter/SenderNameSetting.tsx

```diff
@@ -0,0 +1,72 @@
+import { FormLabel, FormInputValidation } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { ChangeEvent } from 'react';
+import { useSelector } from 'react-redux';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import getSelectedSite from 'calypso/state/ui/selectors/get-selected-site';
+
+type SenderNameSettingProps = {
+	value?: string;
+	disabled?: boolean;
+	replyToValue?: string;
+	updateFields: ( fields: { [ key: string ]: unknown } ) => void;
+};
+
+function replyToExampleEmail( replyToEmail?: string ) {
+	if ( replyToEmail === 'author' ) {
+		return 'author-name@example.com';
+	}
+	if ( replyToEmail === 'comment' ) {
+		return `comment-reply@wordpress.com'`;
+	}
+
+	return 'donotreply@wordpress.com';
+}
+
+export const SenderNameSetting = ( {
+	value = '',
+	disabled,
+	replyToValue,
+	updateFields,
+}: SenderNameSettingProps ) => {
+	const translate = useTranslate();
+	const selectedSite = useSelector( getSelectedSite );
+	const placeholder = selectedSite?.name || translate( 'Sender name' );
```

Comment:
We could fall-back to the domain name here, and also in the backend? 🤔

Reply:
Good idea! Will double check this.